legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -384,31 +386,53 @@ def validate_correction_name_request(filing: Dict, corrected_filing: Dict) -> Op
     return None
 
 
-def validate_name_request(incorporation_filing: Dict) -> Optional[List]:
+def validate_name_request(incorporation_filing: dict, legal_type: str) -> Optional[list]:
     """Validate Name Request section."""
-    nr_path = '/filing/incorporationApplication/nameRequest/nrNumber'
-    nr_number = get_str(incorporation_filing, nr_path)
-
     msg = []
+    nr_path = '/filing/incorporationApplication/nameRequest'
+    nr_number_path = f'{nr_path}/nrNumber'
+    legal_name_path = f'{nr_path}/legalName'
+    legal_type_path = f'{nr_path}/legalType'
+
+    nr_number = get_str(incorporation_filing, nr_number_path)
+    legal_name = get_str(incorporation_filing, legal_name_path)
+    legal_type = get_str(incorporation_filing, legal_type_path)
+
+    valid_numbered_legal_type = [Business.LegalTypes.BCOMP.value,
+                                 Business.LegalTypes.COMP.value,
+                                 Business.LegalTypes.BC_CCC.value,
+                                 Business.LegalTypes.BC_ULC_COMPANY.value]
+    if not nr_number and not legal_name:
+        if legal_type in valid_numbered_legal_type:
+            return None  # It's numbered company
+        else:
+            msg.append({'error': babel('Legal name and nrNumber is missing in nameRequest.'), 'path': nr_path})
+            return msg  # As of today COOP doesn't support numbered company
+
+    if nr_number and not legal_name:
+        msg.append({'error': babel('Legal name is missing in nameRequest.'), 'path': legal_name_path})
+        return msg
+    elif not nr_number and legal_name:
+        msg.append({
+            'error': babel('nrNumber is missing for the legal name provided in nameRequest.'),
+            'path': nr_number_path
+        })
+        return msg
```

Comment:
entity-filer update the legalName available in nameRequest without knowing if nrNumber provided or not. adding these validation to make sure we don't have loophole